File: src/main/java/connectfour/model/Player.java
package connectfour.model;

import connectfour.util.observer.IObserverWithArguments;

public interface Player extends IObserverWithArguments {

    void setName(String string);
    
    String getName();
    
    int getMove();

    public GameField getGameField();
    
    public void setGameField(final GameField gameField);
}

File: src/main/scala/connectfour/model/Player.scala
 package connectfour.model

 import java.awt.Color
 import connectfour.util.observer.IObserverWithArguments;
 
 trait Player extends IObserverWithArguments {
   var name: String
   var color: Color
    
   def getGameField(): GameField
    
   def setGameField(gameField: GameField)
 }